docs/lib/getting-started/fragments/android/integrate.md

```diff
@@ -1,155 +1,459 @@
+            ),
+            todos -> {
+                while (todos.hasNext()) {
+                    Todo todo = todos.next();
+                    Todo updated = todo.copyOfBuilder()
+                            .name("File quarterly taxes")
+                            .build();
+
+                    Amplify.DataStore.save(updated,
+                            success -> Log.i("Tutorial", "Updated item: " + success.item.getName()),
+                            error -> Log.e("Tutorial", "Could not save item to DataStore", error)
+                    );
+                }
+            },
+            failure -> Log.e("Tutorial", "Could not query DataStore", failure)
+    );
+    ```
+
+  </amplify-block>
+
+  <amplify-block name="Kotlin">
+
+    ```kotlin
+    Amplify.DataStore.query(
+            Todo::class.java,
+            Where.matches(
+              Todo.NAME.eq("Finish quarterly taxes")
+            ),
+            Consumer { todos ->
+                while (todos.hasNext()) {
+                    val todo = todos.next()
+                    val updated = todo.copyOfBuilder()
+                            .name("File quarterly taxes")
+                            .build()
+                    Amplify.DataStore.save(updated,
+                            { success -> Log.i("Tutorial", "Updated item: " + success.item.name) },
+                            { error -> Log.e("Tutorial", "Could not save item to DataStore", error) }
+                    )
+                }
+            },
+            Consumer { failure -> Log.e("Tutorial", "Could not query DataStore", failure) }
+    )
+    ```
+
+  </amplify-block>
+  </amplify-block-switcher>
+
+    Models in DataStore are immutable, so in order to update you first create a copy of the object, modify its properties, and then save it.
+
+1. Clear the logcat window by pressing the trashcan icon. Run the application. In logcat, you'll see an indication that the item was updated successfully:
+
+    ```console
+    com.example.todo I/Tutorial: Initialized Amplify
+    com.example.todo I/Tutorial: Updated item: File quarterly taxes
+    ```
+  
+## Delete a Todo
+
+To round out our CRUD operations, we'll query for a record and delete it from DataStore.
+
+1. Edit your `onCreate` method to remove the item creation and save. Your `onCreate()` should only include the code required to initiatize Amplify and not calls to `Todo.builder()` or `Amplify.DataStore.save()`.
+
+1. Below the initialization code, add the following: 
+
+  <amplify-block-switcher>
+  <amplify-block name="Java">
+
+    ```java
+    Amplify.DataStore.query(
+        Todo.class,
+        Where.matches(
+          Todo.NAME.eq("File quarterly taxes")
+        ),
+        todos -> {
+            if (todos.hasNext()) {
+                Todo todo = todos.next();
+
+                Amplify.DataStore.delete(todo,
+                        success -> Log.i("Tutorial", "Deleted item: " + success.item.getName()),
```

Comment:
Curious, do we need to tell customers how to turn logging on?

Reply:
It's automatic. I do tell the customer how to filter the logs, that should be all we need to do.